gulp/tasks/pages.js

```diff
@@ -7,46 +7,41 @@ const glob = require('glob');
 const _ = require('lodash');
 const pageshelpers = require('../utils/pagesHelpers');
 const handleError = require('../utils/handleError');
+const configHelpers = require('../utils/configHelpers');
 
 module.exports = (gulp, $, config) => {
-  const srcFiles = config.appFiles.pages;
-  const destFiles = config.paths.pages.dest;
-  const languages = config.languages;
-  const contentPath = config.paths.content.dest;
-  const baseDir = config.basePaths.src;
-  const manifestFile = config.paths.revManifest.dest;
+  const entryGlob = config.entryGlobs.pages;
+  const destPath = config.destPaths.pages;
+  const contentPath = config.destPaths.content;
+  const definitionGlob = config.entryGlobs.definitions;
+  const manifestDestPath = config.destPaths.revManifest;
 
-  // Put the default language at the root
-  const getLanguagePath = (language) => {
-    if (language === config.languages[0]) {
-      return '';
-    }
-    return `${language}/`;
-  };
-
-  // Returns the relative path between the page and the root of the web server
-  const getRelativePath = (file, language) => {
-    const destPath = config.paths.pages.src + getLanguagePath(language);
-    const filePath = path.dirname(file.path);
-    return `${path.relative(filePath, destPath) || '.'}/`;
-  };
+  // Load the content for the page
+  function loadContentForLanguage(language) {
+    return yamljs.load(`${contentPath}${language}.yml`);
+  }
 
   const task = () => {
-    // Load the content for the page
-    function loadContent(language) {
-      return yamljs.load(`${contentPath}${language}.yml`);
-    }
+    const languages = configHelpers.getAvailableLanguages(config);
 
     function getDestPath(language) {
-      const destPath = destFiles + getLanguagePath(language);
-      return destPath;
+      return destPath + configHelpers.getLanguagePath(language, languages);
     }
 
+    // Returns the relative path between the page and the root of the web server
+    const getRelativePath = (file, language) => {
+      const languageDestPath = config.entryPaths.pages +
+        configHelpers.getLanguagePath(language, languages);
+      const filePath = path.dirname(file.path);
+      return `${path.relative(filePath, languageDestPath) || '.'}/`;
+    };
+
     function loadMergedDefinitions() {
-      return glob.sync(`${config.basePaths.src}**/definition.yml`)
+      return definitionGlob.map(d => glob.sync(d))
```

Comment:
I think in this case is not possible. On the other case we have:

```
glob.sync('*', { cwd: 'content/texts' })
// returns all folders within content/texts
```

```
glob.sync('src/**/definition.yml', { cwd: 'src/' })
// here it doesn't work since the entryGlob also contains the src. So it wouldn't find any file.
```